Create header

cit280superepicness opened the pull request:
Header added to user_home and mostly complete (still need to get Spring Security working so that users can only see "Login" and "Register" on the header when they are looged out and "Logout" when they are logged in. 

Footer complete.

cit280superepicness committed:
added header work-in-progress

cit280superepicness committed:
added header work-in-progress

cit280superepicness committed:
added logo variants and modified header

cit280superepicness committed:
added logo variants and modified header

cit280superepicness committed:
added logo variants and modified header

cit280superepicness committed:
added logo variants and modified header

cit280superepicness committed:
added cloud picture used in logo

cit280superepicness committed:
added cloud picture used in logo

cit280superepicness committed:
updated header

cit280superepicness committed:
updated header

cit280superepicness committed:
added new logo

cit280superepicness committed:
added new logo

cit280superepicness committed:
moved image files to proper folder and fixed footer path in user_home.html

cit280superepicness committed:
moved image files to proper folder and fixed footer path in user_home.html

cit280superepicness committed:
added dependency for spring security

cit280superepicness committed:
added dependency for spring security

cit280superepicness committed:
added line to file to keep browser from demanding username and password to access any web pages

cit280superepicness committed:
added line to file to keep browser from demanding username and password to access any web pages

cit280superepicness committed:
fixed accidental move of files

cit280superepicness committed:
fixed accidental move of files

cit280superepicness committed:
fixed location of index

cit280superepicness committed:
fixed location of index

cit280superepicness committed:
Revert "added line to file to keep browser from demanding username and password to access any web pages"

This reverts commit 26c04ac72d9b1c48d06b786b18a90dac60ca61a2.

cit280superepicness committed:
Revert "added line to file to keep browser from demanding username and password to access any web pages"

This reverts commit 26c04ac72d9b1c48d06b786b18a90dac60ca61a2.

cit280superepicness committed:
Revert "fixed location of index"

This reverts commit 76e1b0a1a5d2df1806efa32b7299f0d7c50c5a89.

cit280superepicness committed:
Revert "fixed location of index"

This reverts commit 76e1b0a1a5d2df1806efa32b7299f0d7c50c5a89.

cit280superepicness committed:
Revert "fixed accidental move of files"

This reverts commit 83d789520300b6a0d30c1729b0ee1a4a6191f15c.

cit280superepicness committed:
Revert "fixed accidental move of files"

This reverts commit 83d789520300b6a0d30c1729b0ee1a4a6191f15c.

cit280superepicness committed:
fixed accidental move of index.html

cit280superepicness committed:
fixed accidental move of index.html

cit280superepicness committed:
added line to stop security from requiring username and password to view pages

cit280superepicness committed:
added line to stop security from requiring username and password to view pages

cit280superepicness committed:
made header into fragment

cit280superepicness committed:
made header into fragment

cit280superepicness committed:
got header working on page. added links to register and login pages

cit280superepicness committed:
got header working on page. added links to register and login pages

cit280superepicness committed:
added method to handle page content depending on if user logged in

cit280superepicness committed:
added method to handle page content depending on if user logged in

cit280superepicness committed:
added code for login and register to be added as fragments

cit280superepicness committed:
added code for login and register to be added as fragments

cit280superepicness committed:
cleanup

cit280superepicness committed:
cleanup

jrmullen commented on pom.xml:
```diff
@@ -57,6 +57,10 @@
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter-security</artifactId>
+        </dependency>
         <!-- end webjar -->
```
remove comment

jrmullen commented:
All images were added to the pull request. Please choose only the images we will be using.

jrmullen commented on pom.xml:
```diff
@@ -57,6 +57,10 @@
             <groupId>org.springframework.boot</groupId>
             <artifactId>spring-boot-starter-jdbc</artifactId>
         </dependency>
+        <dependency>
```
Move security spring dependency outside of the webjar comments

cit280superepicness replied:
Fixed this

jrmullen commented:
I see you say that spring security is not configured. Is this going to cause problems? Have you tested that all pages are functional?

jrmullen commented on src/main/resources/static/index.html:
```diff
@@ -1,134 +0,0 @@
-<!DOCTYPE html>
```
Is this file deleted?

cit280superepicness replied:
No, just empty index.html was empty in my branch for whatever reason. I fixed this. Sorry, I will be more careful from now on with things like this. I was doing a ton of things to try to get the login and register popups working with my header last night and must have committed this by mistake and then not reverted this file when I reverted the rest of my changes.  Sorry about that.

cit280superepicness committed:
fixed issue with extra images in branch

cit280superepicness committed:
fixed issue with extra images in branch

cit280superepicness committed:
delete extra images attempt 2

cit280superepicness committed:
delete extra images attempt 2

cit280superepicness committed:
index.html was empty. fixed this

cit280superepicness committed:
index.html was empty. fixed this

cit280superepicness committed:
Removed dependencies from webjar comments

cit280superepicness committed:
Removed dependencies from webjar comments

cit280superepicness committed:
linked header to index.html

cit280superepicness committed:
linked header to index.html

cit280superepicness commented:
Fixed this.

On Sun, Sep 25, 2016 at 7:01 PM, Jeremy Mullen notifications@github.com
wrote:

> All images were added to the pull request. Please choose only the images
> we will be using.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> https://github.com/jrmullen/Watchdog/pull/14#issuecomment-249452279, or mute
> the thread
> https://github.com/notifications/unsubscribe-auth/AIegPZle55cZus3RkxO80emBnmuKlGJpks5qtv1jgaJpZM4KGBsw
> .

cit280superepicness committed:
Removed this image. We already have one

cit280superepicness committed:
Removed this image. We already have one

cit280superepicness commented:
What I meant by that is that I'm not sure if I have Spring Security 100%
working as intended. If I did not have th:if="${#authorization.
expression('!isAuthenticated()')} commented out in header.html, it would
evaluate to NULL and cause an error. However, it is commented out, so this
is not a problem.

The database errors I am working to resolve (due to my local database not
being configured correctly yet) when I register or login keep me from
viewing the result.html page to ensure it is functional. However, I can say
that the rest of the pages are functional.

On Sun, Sep 25, 2016 at 7:03 PM, Jeremy Mullen notifications@github.com
wrote:

> I see you say that spring security is not configured. Is this going to
> cause problems? Have you tested that all pages are functional?
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> https://github.com/jrmullen/Watchdog/pull/14#issuecomment-249452372, or mute
> the thread
> https://github.com/notifications/unsubscribe-auth/AIegPdcrJGallt3HMa7wILLp2kCb0uz9ks5qtv3egaJpZM4KGBsw
> .

jrmullen commented:
result.html is not associated with database in any way. I don't know what version of the code you have, but if it is the newest version result.html does not exist. It was removed when db was implemented. I only created it to verify that my form was working as intended since I created it before the DB was set up.

So - you either have a DB issue, or a result.html data issue. The two are independent of one another. Check and see which, because if it is a result.html issue then there is a bigger problem then just not having the db configured correctly.

cit280superepicness commented:
Sorry about that. I really need to check over my work a lot more thoroughly
to make sure I don't have any extra files and that I grab the latest remote
master before I create a pull request next time. I did a "git fetch"  and
then "git rebase origin master" to get the latest changes. result.html is
no longer there and I have the latest files now.

I will try to get everything in order regarding the header and user_home,
so we can show it in our demo in class Tuesday.

Renee

On Sun, Sep 25, 2016 at 8:41 PM, Jeremy Mullen notifications@github.com
wrote:

> Reopened #14 https://github.com/jrmullen/Watchdog/pull/14.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> https://github.com/jrmullen/Watchdog/pull/14#event-801656014, or mute
> the thread
> https://github.com/notifications/unsubscribe-auth/AIegPRCLeSZ4mg5bhLh9qYFGRxUnMS6Jks5qtxTRgaJpZM4KGBsw
> .

cit280superepicness committed:
Merge branch 'create_header' of https://github.com/jrmullen/Watchdog into create_header

cit280superepicness commented:
Good news! I got my database set up and register working (it inserts data
into the User table) and I have no result.html anymore.

The bad news is that I am getting a NullPointerException at line 42 of
User.java when I try to login with any user from the database. If this is
not a bug for you as well, I will see where I went wrong (again!) to ensure
that I have the correct files.

 @AssertTrue()

public boolean isDifferentPass() {
    return !passwordConfirm.equals(password) ? false : true;
}

On Sun, Sep 25, 2016 at 10:36 PM, Renee Wise rlwise@oakland.edu wrote:

> Sorry about that. I really need to check over my work a lot more
> thoroughly to make sure I don't have any extra files and that I grab the
> latest remote master before I create a pull request next time. I did a "git
> fetch"  and then "git rebase origin master" to get the latest changes.
> result.html is no longer there and I have the latest files now.
> 
> I will try to get everything in order regarding the header and user_home,
> so we can show it in our demo in class Tuesday.
> 
> Renee
> 
> On Sun, Sep 25, 2016 at 8:41 PM, Jeremy Mullen notifications@github.com
> wrote:
> 
> > Reopened #14 https://github.com/jrmullen/Watchdog/pull/14.
> > 
> > —
> > You are receiving this because you authored the thread.
> > Reply to this email directly, view it on GitHub
> > https://github.com/jrmullen/Watchdog/pull/14#event-801656014, or mute
> > the thread
> > https://github.com/notifications/unsubscribe-auth/AIegPRCLeSZ4mg5bhLh9qYFGRxUnMS6Jks5qtxTRgaJpZM4KGBsw
> > .

jrmullen commented:
The register page is the only page that I set up using DB.

I didn't set up the login page because I thought spring security would take care of that. Is this correct? Anyways the login page is not functional. It will throw errors etc.

As for line 42 it may be a bug and we may have to reconsider. I am using the above code snippet in the register page to confirm that the password and confirm password fields are matching. It may have some weird side effects we will have to look into it further.

jrmullen commented:
Also note that you do not need to most recent code to create a pull request. You just have to know what is functional with what code you have.

cit280superepicness commented:
When you say that you didn't set up the register page, what do you mean
exactly? Did you mean login instead of register?

Spring security should be used to redirect to the Watchdog index.html if
the user is not logged in. This can probably be achieved by storing
credentials and checking that they are valid before loading a page.

On Sep 26, 2016 8:24 AM, "Jeremy Mullen" notifications@github.com wrote:

> The register page is the only page that I set up using DB.
> 
> I didn't set up the register page because I thought spring security would
> take care of that. Is this correct?
> 
> As for line 42 it may be a bug and we may have to reconsider. I am using
> the above code snippet in the register page to confirm that the password
> and confirm password fields are matching. It may have some weird side
> effects we will have to look into it further.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or mute the thread.

jrmullen commented:
login. I made an edit but edits don't show in emails. Sorry.

cit280superepicness commented:
It's no problem. With login, I think we will need it connected to the
database to ensure the user has an accouny. If there is another you have in
mind, please let me know.
I imagine the process would something like make sure the user's credentials
match what is in the database, basically, select \* from user where
user_email and user_password = what the user entered for email and
password. If there is a match in the database, then we will probably set
something in spring security to say that the user is authorized, so they
can access pages like log viewer, user home , etc. Then save the logged in
user's data in an instance of the User class for the duration that they are
logged in. We may use a session object of some type got when the user is
logged in  like in PHP but I'm not sure.

On Sep 26, 2016 1:07 PM, "Jeremy Mullen" notifications@github.com wrote:

> login. I made an edit but edits don't show in emails. Sorry.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> https://github.com/jrmullen/Watchdog/pull/14#issuecomment-249632486, or mute
> the thread
> https://github.com/notifications/unsubscribe-auth/AIegPWBNhMlV5IQWhQiYSvPmjd-IGr8vks5qt_u6gaJpZM4KGBsw
> .

jrmullen commented:
The branch is unable to be merged. There are conflicts in the pom.xml folder. Can you please resolve these?

cit280superepicness commented:
Thanks for the info. I'll take a look. I believe it is because I added a
couple dependencies for Spring Security.

On Mon, Sep 26, 2016 at 8:26 PM, Jeremy Mullen notifications@github.com
wrote:

> The branch is unable to be merged. There are conflicts in the pom.xml
> folder. Can you please resolve these?
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> https://github.com/jrmullen/Watchdog/pull/14#issuecomment-249735396, or mute
> the thread
> https://github.com/notifications/unsubscribe-auth/AIegPbx-vU-QmJDoXY99NrQWyBliq0Bqks5quGK_gaJpZM4KGBsw
> .

jrmullen commented:
IntelliJ should have an to show and resolve conflicts. Google it

cit280superepicness commented:
Still looking into resolving it. I can view the differences in the files in
the Git shell and on IntelliJ and click append changes. However, there is
no indication that clicking append changes anything or if it appends one
line or a few lines of text that are different. The only changes between
the pom.xml files I added a couple lines for the Spring security, added a
comment, and moved the dependency for the Spring JDBC out of the webjars
part of the file.

On Mon, Sep 26, 2016 at 8:54 PM, Jeremy Mullen notifications@github.com
wrote:

> IntelliJ should have an to show and resolve conflicts. Google it
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> https://github.com/jrmullen/Watchdog/pull/14#issuecomment-249739018, or mute
> the thread
> https://github.com/notifications/unsubscribe-auth/AIegPZZ5RVlN93MSlsfojvNaC22P-I-hks5quGlKgaJpZM4KGBsw
> .

cit280superepicness commented:
Nm, I think I found it.

On Mon, Sep 26, 2016 at 9:18 PM, Renee Wise rlwise@oakland.edu wrote:

> Still looking into resolving it. I can view the differences in the files
> in the Git shell and on IntelliJ and click append changes. However, there
> is no indication that clicking append changes anything or if it appends one
> line or a few lines of text that are different. The only changes between
> the pom.xml files I added a couple lines for the Spring security, added a
> comment, and moved the dependency for the Spring JDBC out of the webjars
> part of the file.
> 
> On Mon, Sep 26, 2016 at 8:54 PM, Jeremy Mullen notifications@github.com
> wrote:
> 
> > IntelliJ should have an to show and resolve conflicts. Google it
> > 
> > —
> > You are receiving this because you authored the thread.
> > Reply to this email directly, view it on GitHub
> > https://github.com/jrmullen/Watchdog/pull/14#issuecomment-249739018,
> > or mute the thread
> > https://github.com/notifications/unsubscribe-auth/AIegPZZ5RVlN93MSlsfojvNaC22P-I-hks5quGlKgaJpZM4KGBsw
> > .

cit280superepicness committed:
pulled from master

cit280superepicness commented:
I threw out my pom.xml file and merged with master. I wanted to just append
my changes but unfortunately ran out of time to get it working before I
have to go to bed.

You may have to add the following to pom.xml in origin/master for Spring
Security to work.

```
    <dependency>
        <groupId>org.springframework.boot</groupId>
        <artifactId>spring-boot-starter-security</artifactId>
    </dependency>
```

On Mon, Sep 26, 2016 at 9:23 PM, Renee Wise rlwise@oakland.edu wrote:

> Nm, I think I found it.
> 
> On Mon, Sep 26, 2016 at 9:18 PM, Renee Wise rlwise@oakland.edu wrote:
> 
> > Still looking into resolving it. I can view the differences in the files
> > in the Git shell and on IntelliJ and click append changes. However, there
> > is no indication that clicking append changes anything or if it appends one
> > line or a few lines of text that are different. The only changes between
> > the pom.xml files I added a couple lines for the Spring security, added a
> > comment, and moved the dependency for the Spring JDBC out of the webjars
> > part of the file.
> > 
> > On Mon, Sep 26, 2016 at 8:54 PM, Jeremy Mullen notifications@github.com
> > wrote:
> > 
> > > IntelliJ should have an to show and resolve conflicts. Google it
> > > 
> > > —
> > > You are receiving this because you authored the thread.
> > > Reply to this email directly, view it on GitHub
> > > https://github.com/jrmullen/Watchdog/pull/14#issuecomment-249739018,
> > > or mute the thread
> > > https://github.com/notifications/unsubscribe-auth/AIegPZZ5RVlN93MSlsfojvNaC22P-I-hks5quGlKgaJpZM4KGBsw
> > > .

cit280superepicness commented:
Sorry! I messed up master. I am now trying to revert it. I really don't
know what I'm doing. My apologies about this.

On Mon, Sep 26, 2016 at 9:49 PM, Renee Wise rlwise@oakland.edu wrote:

> I threw out my pom.xml file and merged with master. I wanted to just
> append my changes but unfortunately ran out of time to get it working
> before I have to go to bed.
> 
> You may have to add the following to pom.xml in origin/master for Spring
> Security to work.
> 
> ```
>     <dependency>
>         <groupId>org.springframework.boot</groupId>
>         <artifactId>spring-boot-starter-security</artifactId>
>     </dependency>
> ```
> 
> On Mon, Sep 26, 2016 at 9:23 PM, Renee Wise rlwise@oakland.edu wrote:
> 
> > Nm, I think I found it.
> > 
> > On Mon, Sep 26, 2016 at 9:18 PM, Renee Wise rlwise@oakland.edu wrote:
> > 
> > > Still looking into resolving it. I can view the differences in the files
> > > in the Git shell and on IntelliJ and click append changes. However, there
> > > is no indication that clicking append changes anything or if it appends one
> > > line or a few lines of text that are different. The only changes between
> > > the pom.xml files I added a couple lines for the Spring security, added a
> > > comment, and moved the dependency for the Spring JDBC out of the webjars
> > > part of the file.
> > > 
> > > On Mon, Sep 26, 2016 at 8:54 PM, Jeremy Mullen <notifications@github.com
> > > 
> > > > wrote:
> > > > 
> > > > IntelliJ should have an to show and resolve conflicts. Google it
> > > > 
> > > > —
> > > > You are receiving this because you authored the thread.
> > > > Reply to this email directly, view it on GitHub
> > > > https://github.com/jrmullen/Watchdog/pull/14#issuecomment-249739018,
> > > > or mute the thread
> > > > https://github.com/notifications/unsubscribe-auth/AIegPZZ5RVlN93MSlsfojvNaC22P-I-hks5quGlKgaJpZM4KGBsw
> > > > .

cit280superepicness commented:
Reverted origin/master. I do not want to mess things up further here.
Please just don't include my pom.xml when you merge create_header with
master and it should merge without issues. Then after you merge, to get the
application to compile without errors from Spring Security, please add the
following to origin/master's pom.xml.

   <dependency>
            <groupId>org.springframework.boot</groupId>
            <artifactId>spring-boot-starter-security</artifactId>
        </dependency>

On Mon, Sep 26, 2016 at 9:53 PM, Renee Wise rlwise@oakland.edu wrote:

> Sorry! I messed up master. I am now trying to revert it. I really don't
> know what I'm doing. My apologies about this.
> 
> On Mon, Sep 26, 2016 at 9:49 PM, Renee Wise rlwise@oakland.edu wrote:
> 
> > I threw out my pom.xml file and merged with master. I wanted to just
> > append my changes but unfortunately ran out of time to get it working
> > before I have to go to bed.
> > 
> > You may have to add the following to pom.xml in origin/master for Spring
> > Security to work.
> > 
> > ```
> >     <dependency>
> >         <groupId>org.springframework.boot</groupId>
> >         <artifactId>spring-boot-starter-security</artifactId>
> >     </dependency>
> > ```
> > 
> > On Mon, Sep 26, 2016 at 9:23 PM, Renee Wise rlwise@oakland.edu wrote:
> > 
> > > Nm, I think I found it.
> > > 
> > > On Mon, Sep 26, 2016 at 9:18 PM, Renee Wise rlwise@oakland.edu wrote:
> > > 
> > > > Still looking into resolving it. I can view the differences in the
> > > > files in the Git shell and on IntelliJ and click append changes. However,
> > > > there is no indication that clicking append changes anything or if it
> > > > appends one line or a few lines of text that are different. The only
> > > > changes between the pom.xml files I added a couple lines for the Spring
> > > > security, added a comment, and moved the dependency for the Spring JDBC out
> > > > of the webjars part of the file.
> > > > 
> > > > On Mon, Sep 26, 2016 at 8:54 PM, Jeremy Mullen <
> > > > notifications@github.com> wrote:
> > > > 
> > > > > IntelliJ should have an to show and resolve conflicts. Google it
> > > > > 
> > > > > —
> > > > > You are receiving this because you authored the thread.
> > > > > Reply to this email directly, view it on GitHub
> > > > > https://github.com/jrmullen/Watchdog/pull/14#issuecomment-249739018,
> > > > > or mute the thread
> > > > > https://github.com/notifications/unsubscribe-auth/AIegPZZ5RVlN93MSlsfojvNaC22P-I-hks5quGlKgaJpZM4KGBsw
> > > > > .
